### DISCUSSION TITLE
Bug Report: OpenAI Models Disappear from Model Selection Dropdown

### DISCUSSION THREAD

**Discussion by samuelpetermoshi** [open-webui/open-webui#6822] [Issues]
**Bug Report:**

**Issue Summary:**
Some users of our self-hosted OpenWebUI are experiencing an issue where OpenAI models intermittently disappear from the model selection dropdown box. This problem impacts the user experience and requires manual intervention to resolve temporarily.

**Environment Details:**
- Self-hosted OpenWebUI
- Approximately 250+ users
- Issue observed across multiple user accounts

**Steps to Reproduce:**
The issue appears randomly without specific steps to consistently reproduce it. Users have reported that the OpenAI models sometimes disappear from the dropdown menu during regular usage.

**Temporary Workaround:**
Currently, the workaround is to instruct users to log out and log back in, which causes the models to reappear in the dropdown.

**Additional Notes:**
Any assistance in diagnosing and resolving this issue would be appreciated. Please let us know if further logs or testing information are required. Thank you for looking into this!

**Priority:** High - impacts accessibility for a large number of users.

cc: @ith , @maycszz , @SuvroBaner , @arnab55net 

The Image Below doesn't show the issue stated above it just to point out the OpenAI models 
<img width="1674" alt="Screenshot 2024-11-09 at 23 01 16" src="https://github.com/user-attachments/assets/43a39b3a-229e-438b-bb7e-66c7df948df0">

**Comment by danielj23** [comment]
Where is your ollama machine with the models?  -edit _nm I see you are using the OpenAI setup so my question may not matter._

**Reply by samuelpetermoshi** [reply]
Hello @danielj23,

Thank you for the question! We are currently hosting our Ollama models on an on-premise machine using the OpenWebUI Ollama image. Here's the Docker command we use:

```bash
docker run -d -p ${PORT_MAPPING} --env-file .env --gpus=all -v ollama:/root/.ollama -v ${DATA_VOLUME} --name ${DOCKER_IMAGE} --restart always ghcr.io/open-webui/open-webui:ollama
```

Our OpenWebUI setup includes:
1. **OpenAI Models**
2. **AWS Bedrock Models** (via [AWS Bedrock OpenWebUI pipeline](https://github.com/open-webui/pipelines/blob/main/examples/pipelines/providers/aws_bedrock_claude_pipeline.py))
3. **Google Gemini Models** (via [Google OpenWebUI pipeline](https://github.com/open-webui/pipelines/blob/main/examples/pipelines/providers/google_manifold_pipeline.py))
4. **Ollama Models**

**System Specifications:**
- **Storage**: 4TB+
- **RAM**: 500GB+
- **GPU**: 256GB VRAM
- **CPU**: 80 cores

I hope this answers your question. Feel free to let me know if you need more details.

cc: @ith , @maycszz , @SuvroBaner , @arnab55net

**Comment by arnab55net** [comment]
thank you for reporting this @samuelpetermoshi . this is definitely a problem that is causing bit of headache to our platform users. We hope the Open AI team can take a look at this soon.

**Comment by fsudolsky** [comment]
I noticed the lack of OpenAi after adding pipeline in to the docker compos and adding Anthropics Claude Sonet to Open WebUi but that might be just coincidence

**Comment by topiga** [comment]
I have the exact same issue, although login out and in again doesn't work on my end. Mistral and Anthropic models work, but it's with a pipe function.

**Comment by bdrewery** [comment]
This problem is beyond just the model dropdown. It happens when sending a query to the model as well sometimes, receiving a 404 model not found error. Retrying works. Just now I got it on the summary task after a successful reply.

```
Jun 25 11:52:28 Open-WebUI start.sh[148]: 2025-06-25 11:52:28.951 | INFO     | open_webui.routers.openai:get_all_models:391 - get_all_models() - {}
Jun 25 11:52:28 Open-WebUI start.sh[148]: 2025-06-25 11:52:28.951 | INFO     | open_webui.routers.ollama:get_all_models:334 - get_all_models() - {}
Jun 25 11:52:29 Open-WebUI start.sh[148]: 2025-06-25 11:52:29.388 | INFO     | uvicorn.protocols.http.httptools_impl:send:476 - 10.2.0.65:0 - "GET /api/v1/users/user/settings HTTP/1.1" 200 - {}
Jun 25 11:52:29 Open-WebUI start.sh[148]: 2025-06-25 11:52:29.396 | INFO     | uvicorn.protocols.http.httptools_impl:send:476 - 10.2.0.65:0 - "GET /static/favicon.png HTTP/1.1" 304 - {}
Jun 25 11:52:31 Open-WebUI start.sh[148]: 2025-06-25 11:52:31.085 | INFO     | uvicorn.protocols.http.httptools_impl:send:476 - 10.2.0.65:0 - "POST /api/v1/chats/new HTTP/1.1" 200 - {}
Jun 25 11:52:31 Open-WebUI start.sh[148]: 2025-06-25 11:52:31.090 | INFO     | uvicorn.protocols.http.httptools_impl:send:476 - 10.2.0.65:0 - "GET /static/favicon.png HTTP/1.1" 304 - {}
Jun 25 11:52:31 Open-WebUI start.sh[148]: 2025-06-25 11:52:31.136 | INFO     | uvicorn.protocols.http.httptools_impl:send:476 - 10.2.0.65:0 - "GET /api/v1/chats/?page=1 HTTP/1.1" 200 - {}
Jun 25 11:52:31 Open-WebUI start.sh[148]: 2025-06-25 11:52:31.203 | INFO     | uvicorn.protocols.http.httptools_impl:send:476 - 10.2.0.65:0 - "GET /static/favicon.png HTTP/1.1" 304 - {}
Jun 25 11:52:31 Open-WebUI start.sh[148]: 2025-06-25 11:52:31.227 | INFO     | uvicorn.protocols.http.httptools_impl:send:476 - 10.2.0.65:0 - "POST /api/v1/chats/c2c4c9e3-c1d6-4c0f-b971-f49a0a3b44f7 HTTP/1.1" 200 - {}
Jun 25 11:52:31 Open-WebUI start.sh[148]: 2025-06-25 11:52:31.262 | INFO     | uvicorn.protocols.http.httptools_impl:send:476 - 10.2.0.65:0 - "GET /api/v1/chats/?page=1 HTTP/1.1" 200 - {}
Jun 25 11:52:32 Open-WebUI start.sh[148]: 2025-06-25 11:52:32.340 | ERROR    | open_webui.routers.openai:send_get_request:83 - Connection error:  - {}
Jun 25 11:52:32 Open-WebUI start.sh[148]: 2025-06-25 11:52:32.340 | ERROR    | open_webui.routers.ollama:send_get_request:104 - Connection error:  - {}
Jun 25 11:52:32 Open-WebUI start.sh[148]: 2025-06-25 11:52:32.340 | ERROR    | open_webui.routers.ollama:send_get_request:104 - Connection error:  - {}
Jun 25 11:52:32 Open-WebUI start.sh[148]: 2025-06-25 11:52:32.340 | INFO     | open_webui.routers.openai:get_all_models:391 - get_all_models() - {}
Jun 25 11:52:32 Open-WebUI start.sh[148]: 2025-06-25 11:52:32.344 | INFO     | uvicorn.protocols.http.httptools_impl:send:476 - 10.2.0.65:0 - "GET /ollama/api/version HTTP/1.1" 200 - {}
Jun 25 11:52:33 Open-WebUI start.sh[148]: 2025-06-25 11:52:33.787 | INFO     | uvicorn.protocols.http.httptools_impl:send:476 - 10.2.0.65:0 - "POST /api/chat/completions HTTP/1.1" 200 - {}
Jun 25 11:52:33 Open-WebUI start.sh[148]: 2025-06-25 11:52:33.838 | INFO     | uvicorn.protocols.http.httptools_impl:send:476 - 10.2.0.65:0 - "GET /api/v1/chats/?page=1 HTTP/1.1" 200 - {}
Jun 25 11:52:35 Open-WebUI start.sh[148]: 2025-06-25 11:52:35.343 | ERROR    | open_webui.routers.ollama:send_get_request:104 - Connection error:  - {}
Jun 25 11:52:35 Open-WebUI start.sh[148]: 2025-06-25 11:52:35.451 | INFO     | uvicorn.protocols.http.httptools_impl:send:476 - 10.2.0.65:0 - "GET /api/models HTTP/1.1" 200 - {}
Jun 25 11:52:36 Open-WebUI start.sh[148]: 2025-06-25 11:52:36.179 | INFO     | uvicorn.protocols.http.httptools_impl:send:476 - 10.2.0.65:0 - "GET /_app/immutable/chunks/BKDSxILk.js HTTP/1.1" 304 - {}
Jun 25 11:52:36 Open-WebUI start.sh[148]: 2025-06-25 11:52:36.180 | INFO     | uvicorn.protocols.http.httptools_impl:send:476 - 10.2.0.65:0 - "GET /_app/immutable/chunks/BcMyVBqp.js HTTP/1.1" 304 - {}
Jun 25 11:52:39 Open-WebUI start.sh[148]: 2025-06-25 11:52:39.580 | ERROR    | open_webui.tasks:cleanup_task:86 - Task exception was never retrieved
Jun 25 11:52:39 Open-WebUI start.sh[148]: future: <Task finished name='Task-231103' coro=<process_chat_response.<locals>.post_response_handler() done, defined at /opt/open-webui/backend/open_webui/utils/middleware.py:1373> exception=HTTPException(status_code=404, detail='Model not found')> - >
Jun 25 11:52:39 Open-WebUI start.sh[148]: Traceback (most recent call last):
Jun 25 11:52:39 Open-WebUI start.sh[148]: > File "/opt/open-webui/backend/open_webui/utils/middleware.py", line 2412, in post_response_handler
Jun 25 11:52:39 Open-WebUI start.sh[148]:     await background_tasks_handler()
Jun 25 11:52:39 Open-WebUI start.sh[148]:           └ <function process_chat_response.<locals>.background_tasks_handler at 0x70f59a9cc400>
Jun 25 11:52:39 Open-WebUI start.sh[148]:   File "/opt/open-webui/backend/open_webui/utils/middleware.py", line 1051, in background_tasks_handler
Jun 25 11:52:39 Open-WebUI start.sh[148]:     res = await generate_follow_ups(
Jun 25 11:52:39 Open-WebUI start.sh[148]:                 └ <function generate_follow_ups at 0x70f5a49776a0>
Jun 25 11:52:39 Open-WebUI start.sh[148]:   File "/opt/open-webui/backend/open_webui/routers/tasks.py", line 269, in generate_follow_ups
Jun 25 11:52:39 Open-WebUI start.sh[148]:     raise HTTPException(
Jun 25 11:52:39 Open-WebUI start.sh[148]:           └ <class 'fastapi.exceptions.HTTPException'>
Jun 25 11:52:39 Open-WebUI start.sh[148]: fastapi.exceptions.HTTPException: 404: Model not found
```

**Comment by kinoute** [comment]
I have the same problem with Open WebUI + self hosted mistral model on vLLM instance (using their production stack helm chart).

Hard to find why it randomly happens and in what side the problem is.

**Comment by athoik** [comment]
Have you tried setting [MODELS_CACHE_TTL](https://docs.openwebui.com/getting-started/env-configuration/#models_cache_ttl) to an hour?

**Reply by kinoute** [reply]
@athoik I use (I think) a similar settings in the interface, but it doesn't seem to change. I'm going to look at vLLM.
<img width="2728" height="266" alt="Screenshot 2025-08-15 à 14 58 05" src="https://github.com/user-attachments/assets/835274b9-4d1b-46e3-bff3-1907df4ffb1a" />

**Comment by azzopardim** [comment]
I am seeing a similar behavior. I am running the ollama in a docker container. It gets the latest and reports version 0.11.4. From the terminal I can download and run the oss-gpt. However in the OpenWeb UI the model does not show. I also tried to download and install the model from the OpenWeb UI interface, but it gives an error that the version of Ollama needs to be updated.
